specification/overview.adoc

```diff
@@ -4,128 +4,150 @@
+untrusted resources owned by the non-confidential/hosting supervisor domain, and
+are expected to be donated/transitioned to the confidential supervisor domain
+via ABI supported by the TSM. Once the conversion process is complete,
+confidential memory may be assigned to one or more TVMs by the TSM.
+A converted confidential resource may be freely assigned to another TVM within
+the same supervisor domain when it is no longer in use. However, an
+unused confidential resource must be explicitly reclaimed for use in the
+non-confidential supervisor domain (such memory conversions are tracked and
+enforced via the TSM in the owning supervisor domain to enforce isolation
+properties).
+
+Each TVMs address space can be comprised of confidential and non-confidential
+regions. The former includes both measured pages (that are part of the initial
+TVM payload), and confidential zero-pages that can be mapped-in on demand by
+the VMM following runtime accesses by the TVM. The non-confidential TVM-defined
+regions include those for shared-pages and MMIO.
+
+The TSM implements ABI that are accessed by the OS/VMM in the Hosting Supervisor
+Domain Manager via a *Trusted Execution Environment Interface (TEEI)*. This ABI
+is invoked via a *TSM-Driver* that operates in the M-mode privilege level on the
+hart. The TSM itself operates in HS-mode (priv=01; V=0) of the hart and enables
+the OS/VMM (also in HS-mode) to create TVMs, assign resources to TVMs, manage,
+execute and destroy a TVM - _this specification aims to describe the TEEI and
+TSM interfaces_. By using the Hypervisor extension of the RISC-V privileged
+specification <<R0>>, this specification minimizes ISA changes to introduce
+a scalable architecture for hosting TEE workloads. More than one confidential
+supervisor domains may be hosted by the TSM-driver. Similarly, more than one
+TVMs may be hosted by the host OS/VMM via confidential supervisor domains.
+Each TVM may consist of the guest firmware, a guest OS and applications. The
+software components included in the TVM are implementation specific.
+
+As shown in figure 1, the M-mode firmware TSM-driver is in the TCB of all
+Supervisor domains and hence in the TCB for all CoVE workloads hosted on the
+platform. The TSM-driver (operating in M-mode) uses
 the hardware capabilities to provide:
 
-* Isolation of memory associated with TEEs (including the TSM). We describe 
-*TEE- capable memory* as memory that provides access-control, confidentiality 
-and integrity suitable for use for CoVE components. The TEEI operations for
-memory management are described in detail below. 
+* Isolation of memory associated with TEEs (including the TSM). We describe
+*TEE- capable memory* as memory that is subject to access-control,
+confidentiality and integrity suitable for use for CoVE components.
+For Supervisor Domain memory isolation, the Smmtt extension may be used for
+deployment scenarios where the TSM is at the same privilege level as the VMM.
```

Comment:
Is it better if we align this domain terminology instead of VMM ? For example:
/s/VMM/hosting supervisor domain ?

Reply:
i added a note in the intro section above - The terms hosting supervisor domain and OS/VMM are used interchangeably in this specification.